### PR TITLE
Bugfix immortal_vs

### PR DESCRIPTION
**Pull request by stromblom**
Fixed and refatored to look like the code for op/dp_vs_race perk

### CODE REVIEW

File: src/Calculators/Dominion/CasualtiesCalculator.php [hunk 1]
```diff
@@ -323,4 +322,18 @@ function ($unit) {
...
+     * @param string $opposingForceRaceName
+     * @param int $slot
+     * @return bool
+     */
+    protected function isImmortalVersusRacePerk(Dominion $dominion, string $opposingForceRaceName, int $slot): bool
+    {
+        $raceNameFormatted = strtolower($opposingForceRaceName);
+        $raceNameFormatted = str_replace(' ', '_', $raceNameFormatted);
+
+        return !($dominion->race->getUnitPerkValueForUnitSlot($slot, "immortal_except_vs_{$raceNameFormatted}"));
```
[the remark this answers is not in the document]
Comment (stromblom) [reply src/Calculators/Dominion/CasualtiesCalculator.php]: Do you mean:
```php
return !((bool) $dominion->race->getUnitPerkValueForUnitSlot($slot, "immortal_except_vs_{$raceNameFormatted}"));
```

[the remark this answers is not in the document]
Comment (internetfett) [reply src/Calculators/Dominion/CasualtiesCalculator.php]: Nah, the logic is flipped currently. It's 'isImmortal' so should be returning false most of the time and true only when the perk is found.

[the remark this answers is not in the document]
Comment (stromblom) [reply src/Calculators/Dominion/CasualtiesCalculator.php]: Ah :D I have slept 2 hrs today. :D

[the remark this answers is not in the document]
Comment (internetfett) [reply src/Calculators/Dominion/CasualtiesCalculator.php]: well shit maybe I'm the drunk one

File: src/Calculators/Dominion/CasualtiesCalculator.php [hunk 2]
```diff
@@ -334,6 +334,8 @@ protected function isImmortalVersusRacePerk(Dominion $dominion, string $opposing
         $raceNameFormatted = strtolower($opposingForceRaceName);
         $raceNameFormatted = str_replace(' ', '_', $raceNameFormatted);
```
Comment (WaveHack) [inline src/Calculators/Dominion/CasualtiesCalculator.php]: You can use `snake_case($opposingForceRaceName)` here :) Automatically lowercases and replaced spaces with underscores.

File: src/Calculators/Dominion/CasualtiesCalculator.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (internetfett) [reply src/Calculators/Dominion/CasualtiesCalculator.php]: Now it works!

**Pull request merged**